ios/browser/api/opentabs/brave_opentabs_api.mm

```diff
@@ -0,0 +1,240 @@
+#endif
+
+namespace brave {
+namespace ios {
+sync_pb::SyncEnums::DeviceType SyncDeviceTypeFromSyncEnumsDeviceType(
+    SyncDeviceType deviceType) {
+  switch (deviceType) {
+    case SyncDeviceTypeUnset:
+      return sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_UNSET;
+    case SyncDeviceTypeWin:
+      return sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_WIN;
+    case SyncDeviceTypeMac:
+      return sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_MAC;
+    case SyncDeviceTypeLinux:
+      return sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_LINUX;
+    case SyncDeviceTypeCros:
+      return sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_CROS;
+    case SyncDeviceTypeOther:
+      return sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_OTHER;
+    case SyncDeviceTypePhone:
+      return sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_PHONE;
+    default:
+      return sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_UNSET;
+  }
+}
+
+SyncDeviceType SyncEnumsDeviceTypeFromSyncDeviceType(
+    sync_pb::SyncEnums::DeviceType deviceType) {
+  switch (deviceType) {
+    case sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_UNSET:
+      return SyncDeviceTypeUnset;
+    case sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_WIN:
+      return SyncDeviceTypeWin;
+    case sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_MAC:
+      return SyncDeviceTypeMac;
+    case sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_LINUX:
+      return SyncDeviceTypeLinux;
+    case sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_CROS:
+      return SyncDeviceTypeCros;
+    case sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_OTHER:
+      return SyncDeviceTypeOther;
+    case sync_pb::SyncEnums::DeviceType::SyncEnums_DeviceType_TYPE_PHONE:
+      return SyncDeviceTypePhone;
+    default:
+      return SyncDeviceTypeUnset;
+  }
+}
+}  // namespace ios
+}  // namespace brave
+
+#pragma mark - IOSOpenDistantTab
+
+@implementation IOSOpenDistantTab
+
+- (instancetype)initWithURL:(NSURL*)url
+                      title:(nullable NSString*)title
+                      tabId:(NSInteger)tabId
+                 sessionTag:(NSString*)sessionTag {
+  if ((self = [super init])) {
+    self.url = url;
+    self.title = title;
+    self.tabId = tabId;
+    self.sessionTag = sessionTag;
+  }
+
+  return self;
+}
+
+- (id)copyWithZone:(NSZone*)zone {
+  IOSOpenDistantTab* openDistantTabCopy =
+      [[[self class] allocWithZone:zone] init];
+
+  if (openDistantTabCopy) {
+    openDistantTabCopy.url = self.url;
+    openDistantTabCopy.title = self.title;
+    openDistantTabCopy.tabId = self.tabId;
+    openDistantTabCopy.sessionTag = self.sessionTag;
+  }
+
+  return openDistantTabCopy;
+}
+
+- (void)updateOpenDistantTab:(NSURL*)url title:(NSString*)title {
+  [self setUrl:url];
+
+  if ([title length] != 0) {
```

Comment:
Why we don't update in case of an empty title?

Reply:
Because If title is empty we don't need to update it. Either there is title here already and we don't wanna update it with an empty one in case of some error happened on iOS side or it is already empty, no need to update.